src/chart/graph/categoryVisual.js

```diff
@@ -36,32 +36,19 @@ export default function (ecModel) {
             var color = itemModel.get('itemStyle.color')
                 || seriesModel.getColorFromPalette(name, paletteScope);
             categoriesData.setItemVisual(idx, 'color', color);
+
             var opacity = itemModel.get('itemStyle.opacity') || 1;
             categoriesData.setItemVisual(idx, 'opacity', opacity);
-            var itemSymbolType = itemModel.getShallow('symbol', true);
-            if (itemSymbolType != null) {
-                categoriesData.setItemVisual(idx, 'symbol', itemSymbolType);
-            }
 
-            var itemSymbolSize = itemModel.getShallow('symbolSize', true);
-            if (itemSymbolSize != null) {
-                categoriesData.setItemVisual(idx, 'symbolSize', itemSymbolSize);
-            }
+            var symbolStyleList = ['symbol','symbolSize','symbolKeepAspect','symbolOffset','symbolRotate']
 
-            var itemSymbolKeepAspect = itemModel.getShallow('symbolKeepAspect',true);
-            if (itemSymbolKeepAspect != null) {
-                categoriesData.setItemVisual(idx, 'symbolKeepAspect', itemSymbolKeepAspect);
-            }
-
-            var itemSymbolOffset = itemModel.getShallow('symbolOffset',true);
-            if (itemSymbolOffset != null) {
-                categoriesData.setItemVisual(idx, 'symbolOffset', itemSymbolOffset);
+            for(var i = 0;i < symbolStyleList.length;i++){
+                var symbolStyleItem = itemModel.getShallow(symbolStyleList[i], true);
```

Comment:
Did you test `graph-symbol.html`? It seems `itemModel.getShallow('symbolOffset', true)` always return undefined, so that `symbolOffset: ['30%', 0]` is config is not used. This doesn't seem right.

Reply:
Both symbolOffset and other items return undefined, but somehow it works.
In my local server, change the symbolOffset do bring some effects to symbol. You could try to put the percentage larger to see that, but I am not sure whether the position is right.
Sorry for those mistakes I have made... Here is a new version with formatted code.

Reply:
Reduct the latter symbolStyleList with symbolOffset and symbolRotate, it also works. Others need both two function to do that.
In fact I don't know why...

Reply:
Well, we have to make sure we understand our changes. 🤣 
I'll check it later. Thanks!